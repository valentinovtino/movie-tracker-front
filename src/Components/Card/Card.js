import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Redirect } from 'react-router-dom';

class Card extends Component {
  constructor(props) {
    super(props)

    this.state = {
      hasErrored: false
    }
  }
  
  handleFavorite = () => {  
    if (!this.props.userID) {
      this.setState({ hasErrored: true })
    } else {
      this.props.addUserFavorite(this.props.movie, this.props.userID);
    }
  };

  render () {
    const errorRedirect =
      this.state.hasErrored === true ?
        <Redirect to='/login' /> :
        <div></div>;

    return (
      <div className='flip-container'>
        {errorRedirect}
        <div className='flipper'>
          <div className='Card front'>
            <img src={`https://image.tmdb.org/t/p/w500${this.props.movie.posterPath}`} />
          </div>
          <div className='Card back'>
            <h1>{this.props.movie.title}</h1>
            <h3>Rating: {this.props.movie.averageRating}</h3>
            <button onClick={this.handleFavorite}> Add to favorites </button>
            <h3>{this.props.movie.releaseData}</h3>
            <p>{this.props.movie.overview}</p>
          </div>
        </div>
      </div>
    );
  }
}

Card.propTypes = {
  movie: PropTypes.object.isRequired
};

export default Card;