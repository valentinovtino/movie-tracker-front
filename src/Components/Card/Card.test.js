import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card'

describe('Card', () => {
  it('should render a card', () => {
    const wrapper = shallow(<Card />);

    expect(wrapper).toMatchSnapshot();
  });
});