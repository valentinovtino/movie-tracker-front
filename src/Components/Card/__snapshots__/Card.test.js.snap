// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render a card 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="flipper"
>
        <div
                className="Card front"
        >
                <img
                        src="https://image.tmdb.org/t/p/w500undefined"
                />
        </div>
        <div
                className="Card back"
        >
                <h1 />
                <h3>
                        Rating: 
                </h3>
                <h3 />
                <p />
        </div>
</div>,
      "className": "flip-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="Card front"
>
            <img
                        src="https://image.tmdb.org/t/p/w500undefined"
            />
</div>,
          <div
            className="Card back"
>
            <h1 />
            <h3>
                        Rating: 
            </h3>
            <h3 />
            <p />
</div>,
        ],
        "className": "flipper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              src="https://image.tmdb.org/t/p/w500undefined"
/>,
            "className": "Card front",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "src": "https://image.tmdb.org/t/p/w500undefined",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1 />,
              <h3>
                Rating: 
</h3>,
              <h3 />,
              <p />,
            ],
            "className": "Card back",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Rating: ",
                  undefined,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Rating: ",
                undefined,
              ],
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="flipper"
>
          <div
                    className="Card front"
          >
                    <img
                              src="https://image.tmdb.org/t/p/w500undefined"
                    />
          </div>
          <div
                    className="Card back"
          >
                    <h1 />
                    <h3>
                              Rating: 
                    </h3>
                    <h3 />
                    <p />
          </div>
</div>,
        "className": "flip-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="Card front"
>
              <img
                            src="https://image.tmdb.org/t/p/w500undefined"
              />
</div>,
            <div
              className="Card back"
>
              <h1 />
              <h3>
                            Rating: 
              </h3>
              <h3 />
              <p />
</div>,
          ],
          "className": "flipper",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                src="https://image.tmdb.org/t/p/w500undefined"
/>,
              "className": "Card front",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "src": "https://image.tmdb.org/t/p/w500undefined",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1 />,
                <h3>
                  Rating: 
</h3>,
                <h3 />,
                <p />,
              ],
              "className": "Card back",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Rating: ",
                    undefined,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Rating: ",
                  undefined,
                ],
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
